ipykernel/kernelapp.py

```diff
@@ -437,13 +449,13 @@ def init_signal(self):
     def init_kernel(self):
         """Create the Kernel object itself"""
         shell_stream = ZMQStream(self.shell_socket)
-        control_stream = ZMQStream(self.control_socket)
+        control_stream = ZMQStream(self.control_socket, self.control_thread.io_loop)
 
         kernel_factory = self.kernel_class.instance
 
         kernel = kernel_factory(parent=self, session=self.session,
                                 control_stream=control_stream,
-                                shell_streams=[shell_stream, control_stream],
```

Comment:
Can we leave shell_streams as a list? ipyparallel relies on this API and even if the default length is 1, there's no harm in preserving the type for compatibility. It should also make this patch smaller, since there's no type change to work around.

Reply:
Okay, it makes sense if this is an API that people depend upon.

(I thought it was simplifying the code to drop the list).

Reply:
I added a deprecated `shell_streams` list property. Let me know if you think that it is fine.